src/utils.ts

```diff
@@ -0,0 +1,35 @@
+import cbor from 'cbor'
+
+import type { RewardAccount} from './types'
+import { StakeCredentialType } from './types'
+
+export const decodeCbor = (buffer: Buffer) => cbor.decode(buffer, {
+    allowMapsWithDuplicateKeys: false,
+    tags: {
+        // Specifies that tag 30 should be parsed only as a tuple. For example
+        // the CDDL specifies unit_interval as:
+        // unit_interval = #6.30([uint, uint])
+        30: (v: any) => {
+            if (!Array.isArray(v) || v.length != 2) {
+                throw new Error('Invalid tuple')
+            }
+            return v
+        },
+    },
+})
+
+export const encodeToCbor = (x: any) => cbor.encodeOne(x, {canonical: true})
+
+export const bind = <A, R, T extends any[]>(fn: (x: A, ...args: T) => R, ...args: T): (x: A) => R => (x: A) => fn(x, ...args)
+
+export const rewardAccountStakeCredentialType = (rewardAccount: RewardAccount) => {
+    switch(rewardAccount[0] >> 4) {
```

Comment:
I think this should actually be
```suggestion
    switch((rewardAccount[0] >> 4) & 1) {
```
Or the case values should be adjusted to 14 and 15.